3rdparty/libtim-vx/tim-vx.cmake

```diff
@@ -0,0 +1,53 @@
+set(TIMVX_COMMIT_HASH "e001d53ddfe1df86838d2bc593d043358e8a4527")
+set(OCV_TIMVX_DIR "${OpenCV_BINARY_DIR}/3rdparty/libtim-vx")
+set(OCV_TIMVX_SOURCE_PATH "${OCV_TIMVX_DIR}/TIM-VX-${TIMVX_COMMIT_HASH}")
+
+if(EXISTS "${OCV_TIMVX_SOURCE_PATH}")
+    message(STATUS "Tim-VX exists already at ${OCV_TIMVX_SOURCE_PATH}")
+
+    set(TIMVX_FOUND ON)
+    set(BUILD_TIMVX ON)
+else()
+    set(OCV_TIMVX_FILENAME "${TIMVX_COMMIT_HASH}.zip")
+    set(OCV_TIMVX_URL "https://github.com/VeriSilicon/TIM-VX/archive/")
+    set(timvx_zip_md5sum 4e7bdd168d76e6937096132ca430791d)
+
+    ocv_download(FILENAME ${OCV_TIMVX_FILENAME}
+                 HASH ${timvx_zip_md5sum}
+                 URL "${OCV_TIMVX_URL}"
+                 DESTINATION_DIR "${OCV_TIMVX_DIR}"
+                 ID TIMVX
+                 STATUS res
+                 UNPACK RELATIVE_URL)
+
+    if(NOT res)
+        set(TIMVX_FOUND OFF)
+        message(STATUS "Failed to download Tim-VX source code from github. Turning off TIMVX_FOUND")
+    else()
+        set(TIMVX_FOUND ON)
+        set(BUILD_TIMVX ON)
+        message(STATUS "TIMVX: Downloaded Tim-VX.")
+    endif()
+endif()
+
+if(BUILD_TIMVX)
+    set(HAVE_TIMVX 1)
+
+    ocv_warnings_disable(CMAKE_C_FLAGS -Wunused-parameter -Wstrict-prototypes -Wundef -Wsign-compare -Wmissing-prototypes -Wmissing-declarations -Wstrict-aliasing -Wunused-but-set-variable -Wmaybe-uninitialized -Wshadow -Wsuggest-override)
+    ocv_warnings_disable(CMAKE_CXX_FLAGS -Wunused-parameter -Wstrict-prototypes -Wundef -Wsign-compare -Wunused-but-set-variable -Wshadow -Wsuggest-override -Wmissing-declarations)
+
+    set(BUILD_IN_OPENCV ON)
+    set(TimVX_INCLUDE_DIR "${OCV_TIMVX_SOURCE_PATH}/include" CACHE INTERNAL "")
+    if(EXISTS "${OCV_TIMVX_SOURCE_PATH}/CMakeLists.txt")
+        set(BUILD_SHARED_LIBS OFF CACHE "" INTERNAL FORCE)
+        add_subdirectory("${OCV_TIMVX_SOURCE_PATH}" "${OCV_TIMVX_DIR}/build")
+        set(BUILD_SHARED_LIBS ON CACHE "" INTERNAL FORCE)
```

Comment:
Do we have a cleaner way to override the `BUILD_SHARED_LIBS` var in TIMVX's CMakeLists.txt?

Reply:
Fixed with replacing source of TIM-VX to my own fork. Also `ocv_install` is moved to my fork.